src/content/en/updates/posts/2016/05/houdini.md

```diff
@@ -0,0 +1,366 @@
+make many of the later drafts possible. If you thought of Web Workers when you
+read “worklet”, you are not wrong. They have a lot of conceptual overlap. So why
+a new thing when we already have workers? Houdini’s goal is to expose new APIs
+to allow web developers to hook up their own code into the CSS engine and the
+surrounding systems. It’s probably not unrealistic to assume that some of these
+code fragments will have to be run *every. single. frame*. Some of them have to
+by definition. Quoting the [Web Worker spec]:
+
+> Workers [...] are relatively heavy-weight, and are not intended to be used in
+large numbers. For example, it would be inappropriate to launch one worker for
+each pixel of a four megapixel image.
+
+That means web workers are not viable for the things Houdini plans to do.
+Therefore, worklets were invented. Worklets make use of ES2015 classes as nice
+way to define a collection of methods, the signatures of which are predefined by
+the type of the worklet. They are light-weight and short-lived.
+
+### Paint Worklet ([spec][Paint Worklet spec])
+
+I am starting with this as it is introduces the fewest new concepts. From the
+spec draft itself:
+
+> The paint stage of CSS is responsible for painting the background, content and
+highlight of an element based on that element’s geometry (as generated by the
+layout stage) and computed style.
+
+This allows you not only to define how an element should draw itself (think of
+the synergy with Web Components!) but also *alter* the visuals of existing
+elements. No need for hacky things like DOM elements to create a ripple effect
+on buttons. This would allow a significant reduction in DOM node count for
+common visuals. Another big advantage of running your code at paint time of an
+element compared to using a regular canvas is that you will know the size of the
+element you are supposed to paint and that you will be aware of fragments and
+handle them appropriately.
+
+Wait, what are fragments?
+
+#### Fragments
+
+I think of elements in the DOM tree as boxes that are laid out by the CSS engine
+to make up my website. It turns out, however, that this mental model becomes
+flawed once inline elements come into play. A `<span>` may need to be wrapped;
+so while still technically being a single DOM node, it has been *fragmented*
+into 2, well, *fragments*. The [spec][Fragmentation spec] calls the bounding box
+of these 2 fragments a *fragmentainer*. I am not even kidding.
+
+<img src="/web/updates/images/2016/05/houdini/fragment.png">
+
+Back to the Paint Worklet: Effectively, your code will get called for each
+fragment and will be given access to a stripped down `<canvas>`-like API as well
+as the styles applied to the element, which allows you to draw (but not visually
+inspect) the fragment. You can even request an “overflow” margin to allow you to
+draw effects *around* the element’s boundaries, just like `box-shadow`.
+
+{% highlight JS %}
```

Comment:
This code is presented without much context—is it an implementation of `box-shadow` in a Paint Worklet? Can you add a title or an explanatory comment to clarify?